vm/systemSmartContracts/delegationManager.go

```diff
@@ -495,6 +499,92 @@ func (d *delegationManager) getContractConfig(args *vmcommon.ContractCallInput)
 	return vmcommon.Ok
 }
 
+func (d *delegationManager) claimMulti(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	returnCode := d.executeFuncOnListAddresses(args, claimRewards)
+	if returnCode != vmcommon.Ok {
+		return returnCode
+	}
+	totalSent := d.eei.GetTotalSentToUser(args.CallerAddr)
+	d.eei.Finish(totalSent.Bytes())
+
+	return vmcommon.Ok
+}
+
+func (d *delegationManager) reDelegateMulti(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	returnCode := d.executeFuncOnListAddresses(args, reDelegateRewards)
+	if returnCode != vmcommon.Ok {
+		return returnCode
+	}
+	logs := d.eei.GetLogs()
+	totalReDelegated := getTotalReDelegatedFromLogs(logs)
+	d.eei.Finish(totalReDelegated.Bytes())
+
+	return vmcommon.Ok
+}
+
+func getTotalReDelegatedFromLogs(logs []*vmcommon.LogEntry) *big.Int {
+	totalReDelegated := big.NewInt(0)
+	for _, reDelegateLog := range logs {
+		if len(reDelegateLog.Topics) < 1 {
+			continue
+		}
+		if !bytes.Equal(reDelegateLog.Identifier, []byte(delegate)) {
+			continue
+		}
+		valueFromFirstTopic := big.NewInt(0).SetBytes(reDelegateLog.Topics[0])
+		totalReDelegated.Add(totalReDelegated, valueFromFirstTopic)
+	}
+
+	return totalReDelegated
+}
+
+func (d *delegationManager) executeFuncOnListAddresses(
+	args *vmcommon.ContractCallInput,
+	funcName string,
+) vmcommon.ReturnCode {
+	if !d.enableEpochsHandler.IsMultiClaimOnDelegationEnabled() {
+		d.eei.AddReturnMessage("invalid function to call")
```

Comment:
Why we don't have this error and the one in L569 defined like the others? Ex: Like the one in L550 -> vm.ErrInvalidNumOfArguments.Error()

Reply:
Because of duplicated code.

Reply:
Because of backward compatibility